### DISCUSSION TITLE
Full-text search on trace/observation/generation input/output

### DISCUSSION THREAD

**Discussion by marcklingen** [langfuse/langfuse#939] [Ideas]
It'd be useful to search the input/output of objects in Langfuse to find conversations that include a specific tool, intent, context, ...

The naive implementation via Postgres LIKE/ILIKE is very slow given the size of the trace/observation tables in Langfuse production deployments.

Let's gage the overall interest for this feature and investigate potential solutions (better indexing, move to OLAP, optional third-party full-text search).

**Update, now live: https://langfuse.com/changelog/2026-05-27-clickhouse-full-text-search-fast-mode**

**Comment by arnsenglander** [comment]
Yes pls!

**Comment by aiakubovich** [comment]
Interested as well

**Comment by zuzaai** [comment]
Very interested as well!

**Comment by marcklingen** [comment]
agree, high up on the v3-wishlist :)

**Comment by p-barman** [comment]
Bumping! Would love this feature :)

**Reply by marcklingen** [reply]
thank you, I'd love to see this as well :)

**Comment by lshalon** [comment]
Interested! I think that REGEX should be fairly fast if you do it time-gated :)

**Comment by n2nco** [comment]
Super needed - personally can't migrate to langfuse w/o it

**Reply by felixkrrr** [reply]
Hey @n2nco, it's Felix from Langfuse. Not sure if you have seen it but full-text search is live as of last week! Check out the launch [here](https://langfuse.com/changelog/2025-05-19-full-text-search), let us know what you think!

**Comment by vigneshmj1997** [comment]
Much Interested As well

**Comment by hammond756** [comment]
Would love this! That way I can quickly navigate to generations of interest or weird cases identified by users

**Reply by marcklingen** [reply]
agree! btw, you could use your own ids as a trace id to be able to deeplink to traces in langfuse based on user reports. more on this here: https://langfuse.com/docs/tracing-features/url

**Comment by thomasnormal** [comment]
Right now my workaround is:

- Download all traces as a huge json file
- Don't try to open the json file in an editor (it will crash as everything is on a single line)
- Instead use ipython to parse the file and loop through all the traces, searching for the string...

**Comment by anatolec** [comment]
If you do this please make it possible to search by JSON-path in structured output 🙏

**Comment by SinghCoder** [comment]
This would be a huge + in experience. Looking forward to this.

**Comment by kincal** [comment]
🙏

**Comment by ChristianP89** [comment]
+1 this would be super useful

**Comment by rodrigocpaiva** [comment]
+1 this would be super useful

**Comment by skabbit** [comment]
Still looking for this feature to help finding specific cases in UI. Should super helpful for composing datasets.

**Comment by Haoping-Xiao** [comment]
@marcklingen any progresses or plans?

**Comment by marcklingen** [comment]
This is on the roadmap, will post an update here once an initial version is available

**Comment by marcklingen** [comment]
@maxdeichmann just released an initial version of full-text search. Please share any feedback you might have here.

Learn more: https://langfuse.com/changelog/2025-05-09-full-text-search

**Reply by SinghCoder** [reply]
does this require to be on v3.*+?

**Reply by maxdeichmann** [reply]
> does this require to be on v3.*+?

Yes, this is not available on V2.

**Reply by amargitx** [reply]
link looks broken, works for me https://langfuse.com/blog/2025-05-19-launch-week-3

**Comment by uahmad235** [comment]
Is it available on the self-hosted OSS version?

**Comment by clemra** [comment]
Big update to this today: https://langfuse.com/changelog/2026-05-27-clickhouse-full-text-search-fast-mode
